servers/src/main/java/tachyon/worker/block/BlockMasterSync.java

```diff
@@ -220,4 +220,33 @@ private void resetMasterClient() {
         new MasterClient(NetworkAddressUtils.getMasterAddress(mTachyonConf),
             mMasterClientExecutorService, mTachyonConf);
   }
+
+  /**
+   * Thread to remove block from master
+   */
+  private class BlockRemover implements Runnable {
+    private BlockDataManager mBlockDataManager;
+    private long mUserId;
+    private long mBlockId;
+
+    public BlockRemover(BlockDataManager blockDataManager, long userId, long blockId) {
+      mBlockDataManager = blockDataManager;
+      mUserId = userId;
+      mBlockId = blockId;
+    }
+
+    @Override
+    public void run() {
+      try {
+        mBlockDataManager.removeBlock(mUserId, mBlockId);
+      } catch (IOException ioe) {
```

Comment:
Thanks for your reply, I do understand the try-catch mechanism in JAVA, I proposed that question just for discussion, all is up to you here :)